MantidQt/MantidWidgets/src/InstrumentView/ObjComponentActor.cpp

```diff
@@ -50,14 +48,25 @@ namespace MantidWidgets {
 
 ObjComponentActor::ObjComponentActor(const InstrumentActor &instrActor,
                                      Mantid::Geometry::ComponentID compID)
-    : ComponentActor(instrActor, compID) {
+    : ComponentActor(instrActor, compID), m_log("ObjComponentActor"),
+      g_log(m_log) {
   // set the displayed colour
   setColors();
 
-  if (!isComponentFinite(compID)) {
-    // If the component does not have finite length we set it always
-    // hidden so scale is not messed up and it is not displayed.
+  if (compID->hasParameter(ParameterMap::isHidden()) &&
```

Comment:
Confused. You are not using `ComponentInfo` despite the fact that you have the information there.